cdap-ui/app/cdap/components/ConfigurationGroup/utilities/DynamicPluginFilters.ts

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+// jexl library to parse js expression specified in widget json
+import jexl from 'jexl';
+import {
+  IConfigurationGroup,
+  IWidgetJson,
+  PluginProperties,
+  IWidgetProperty,
+  IPropertyFilter,
+  CustomOperator,
+  IPropertyValues,
+  IPropertyValueType,
+  IPropertyTypedValues,
+} from 'components/ConfigurationGroup/types';
+import flatten from 'lodash/flatten';
+import difference from 'lodash/difference';
+import { objectQuery } from 'services/helpers';
+
+interface IFilteredWidgetProperty extends IWidgetProperty {
+  show?: boolean;
+}
+
+export interface IFilteredConfigurationGroup extends IConfigurationGroup {
+  show?: boolean;
+  properties: IFilteredWidgetProperty[];
+}
+
+function isValueMacro(value) {
```

Comment:
there is already an isMacro function under utilities. Can it be reused or consolidated? The function is slightly different

Reply:
Didn't realize we already had one. Have moved the existing one to `services/helpers` and changed all references including this one.